nexus/src/app/address_lot.rs

```diff
@@ -19,23 +22,43 @@ use std::sync::Arc;
 use uuid::Uuid;
 
 impl super::Nexus {
+    pub fn address_lot_lookup<'a>(
+        &'a self,
+        opctx: &'a OpContext,
+        address_lot: NameOrId,
+    ) -> LookupResult<lookup::AddressLot<'a>> {
+        match address_lot {
+            NameOrId::Id(id) => {
+                let lot = LookupPath::new(opctx, &self.db_datastore)
+                    .address_lot_id(id);
+                Ok(lot)
+            }
+            NameOrId::Name(name) => {
+                let lot = LookupPath::new(opctx, &self.db_datastore)
+                    .address_lot_name_owned(name.into());
+                Ok(lot)
+            }
+        }
+    }
+
     pub async fn address_lot_create(
         self: &Arc<Self>,
         opctx: &OpContext,
         params: params::AddressLotCreate,
     ) -> CreateResult<AddressLotCreateResult> {
+        opctx.authorize(authz::Action::CreateChild, &authz::FLEET).await?;
         validate_blocks(&params)?;
-        opctx.authorize(authz::Action::Modify, &authz::FLEET).await?;
         self.db_datastore.address_lot_create(opctx, &params).await
     }
 
     pub async fn address_lot_delete(
         self: &Arc<Self>,
         opctx: &OpContext,
-        name_or_id: &Option<NameOrId>,
+        address_lot_lookup: &lookup::AddressLot<'_>,
```

Comment:
As mentioned above, I've swapped out the `name_or_id` reference for a lookup. That lookup makes it easy for us to do authz checks on it like you'll see below.